src/Tests/dotnet-workload-install.Tests/GivenDotnetWorkloadInstall.cs

```diff
@@ -293,21 +304,48 @@ public void GivenWorkloadInstallItDoesNotRemoveOldInstallsOnRollback()
             var testDirectory = _testAssetsManager.CreateTestDirectory(testName: testName).Path;
             var dotnetRoot = Path.Combine(testDirectory, "dotnet");
             var installer = new MockPackWorkloadInstaller(failingWorkload);
-            var workloadResolver = WorkloadResolver.CreateForTests(new MockManifestProvider(new[] { _manifestPath }), new string[] { dotnetRoot });
+            var workloadResolver = WorkloadResolver.CreateForTests(new MockManifestProvider(new[] { _manifestPath }), dotnetRoot);
             var nugetDownloader = new MockNuGetPackageDownloader(dotnetRoot);
             var manifestUpdater = new MockWorkloadManifestUpdater(manifestUpdates, tempDirManifestPath);
+            string sdkVersion = "6.0.100";
+            if (_userLocal)
+            {
+                SetUserLocal(dotnetRoot, sdkVersion);
+            }
             var installManager = new WorkloadInstallCommand(
                 parseResult,
                 reporter: _reporter,
                 workloadResolver: workloadResolver,
                 workloadInstaller: installer,
                 nugetPackageDownloader: nugetDownloader,
                 workloadManifestUpdater: manifestUpdater,
-                userHome: testDirectory,
+                userProfileDir: testDirectory,
                 dotnetDir: dotnetRoot,
                 version: "6.0.100");
 
             return (testDirectory, installManager, installer, workloadResolver, manifestUpdater, nugetDownloader);
         }
+
+        private void SetUserLocal(string dotnetDir, string sdkFeatureBand)
+        {
+            // Add the userlocal marker file.
+            string workloadMetadataDir = Path.Combine(dotnetDir, "metadata", "workloads", sdkFeatureBand);
+            Directory.CreateDirectory(workloadMetadataDir);
+            File.WriteAllText(Path.Combine(workloadMetadataDir, "userlocal"), "");
+
+            Assert.True(WorkloadInstall.IsUserLocal(dotnetDir, sdkFeatureBand));
+        }
+    }
+
+    public class UserLocalGivenDotnetWorkloadInstall : GivenDotnetWorkloadInstallBase
+    {
+        public UserLocalGivenDotnetWorkloadInstall(ITestOutputHelper log) : base(log, userLocal: true)
```

Comment:
With this approach it looks like we'll end up running tests twice, even if they're not set up to be user local, in which case we'd have needless duplication. I'd prefer to avoid making GivenDotnetWorkloadInstall abstract and instead make a few tests theories (the data being true or false for user local).